gobblin-core/src/main/java/gobblin/writer/http/AbstractHttpWriterBuilder.java

```diff
@@ -1,32 +1,132 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
 package gobblin.writer.http;
 
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import lombok.Getter;
+
+import org.apache.http.client.config.RequestConfig;
 import org.apache.http.conn.HttpClientConnectionManager;
 import org.apache.http.impl.client.HttpClientBuilder;
 import org.apache.http.impl.conn.BasicHttpClientConnectionManager;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
 import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
 
 import gobblin.config.ConfigBuilder;
 import gobblin.configuration.State;
+import gobblin.writer.Destination;
 import gobblin.writer.FluentDataWriterBuilder;
 
-
 public abstract class AbstractHttpWriterBuilder<S, D, B extends AbstractHttpWriterBuilder<S, D, B>>
     extends FluentDataWriterBuilder<S, D, B> {
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractHttpWriterBuilder.class);
+
   public static final String CONF_PREFIX = "gobblin.writer.http.";
+  public static final String HTTP_CONN_MANAGER = "conn_mgr_type";
```

Comment:
AbstractHttpWriterBuilder uses gobblin.config.ConfigBuilder where ConfigBuilder strips out the prefix from config key. Thus, config keys do not have CONF_PREFIX.